### PR TITLE
TD0008 filter task

### PR DESCRIPTION
**Pull request by amit-ib**

### CODE REVIEW

File: src/components/todo/TodoList.tsx [hunk 1]
```diff
@@ -5,13 +5,30 @@ import TodoItem from "./TodoItem";
...
   todos: TodoModal[];
   setTodos: React.Dispatch<React.SetStateAction<TodoModal[]>>; // copied from setTodos state
+  setFilter?: React.Dispatch<React.SetStateAction<string>>; // copied from setTodos state
+  filter?: string;
 }
 
-const TodoList = ({ todos, setTodos }: Props) => {
+const TodoList = ({ todos, setTodos, filter }: Props) => {
+  let filteredList = todos;
+  if (filter === "done") {
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoList.tsx:14]: Try to use ENUM here

Comment (amit-ib) [reply src/components/todo/TodoList.tsx:14]: Updated and used ENUM

File: src/assets/scss/_common.scss [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
   background: #3f94ff;
 }
-
+h3 {
+  font-size: 1.17em;
+  margin: 1em 0px;
+  font-weight: bold;
+}
 .container {
   margin: 20px auto 20px auto;
```
Comment (akash-dhande-ib) [inline src/assets/scss/_common.scss:15]: Either use 'EM' or 'PX'

File: src/assets/scss/components/_list-item.scss [hunk 3]
```diff
@@ -5,18 +5,29 @@
...
     display: flex;
-    span {
+    .button {
       display: flex;
       margin: 0px 5px;
       font-size: 12px;
       align-items: center;
       cursor: pointer;
+      background: transparent;
+      padding: 0px;
```
Comment (akash-dhande-ib) [inline src/assets/scss/components/_list-item.scss:30]: Redundant styles for button.

File: public/index.html [hunk 4]
```diff
@@ -39,5 +40,9 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
-  </body>
-</html>
+  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js"
+    integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM"
+    crossorigin="anonymous"></script>
+</body>
```
Comment (akash-dhande-ib) [inline public/index.html:46]: Avoid using this.

File: src/App.tsx [hunk 5]
```diff
@@ -26,12 +26,15 @@ function App() {
...
     setIsLogedin(accessToken ? true : false);
-    console.log("isLogedin", isLogedin, accessToken);
   }, []);
 
   const [todos, setTodos] = useState<TodoModal[]>(staticTodo);
   const [userProfile, setuserProfile] = useState<UserModal | null>();
 
+  // state to handle filters
+  const [filter, setFilter] = useState<string>("");
+  console.log(filter);
```
Comment (akash-dhande-ib) [inline src/App.tsx:36]: Remove debug

File: src/components/todo/TodoItem.tsx [hunk 6]
```diff
@@ -1,46 +1,171 @@
...
-    if (todos[index]) {
-      todos[index].isDone = !todos[index].isDone;
-    }
+    todos.forEach((item) => {
+      if (item.id === id) {
+        item.isDone = !item.isDone;
+      }
+    });
+    setTodos(todos);
+  };
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:50]: It would cost performance loss

**Review by akash-dhande-ib — CHANGES_REQUESTED** [review]
Try to create small segregated PR

File: src/App.tsx (hunk 5 above)
Comment (amit-ib) [reply src/App.tsx:36]: Debugging removed

File: public/index.html (hunk 4 above)
Comment (amit-ib) [reply public/index.html:46]: used react-bootstrap instead

File: src/assets/scss/components/_list-item.scss (hunk 3 above)
Comment (amit-ib) [reply src/assets/scss/components/_list-item.scss:30]: updated

File: src/assets/scss/_common.scss (hunk 2 above)
Comment (amit-ib) [reply src/assets/scss/_common.scss:15]: updated

Comment (akash-dhande-ib) [reply src/assets/scss/_common.scss:15]: I don't see the changes

File: public/index.html (hunk 4 above)
Comment (akash-dhande-ib) [reply public/index.html:46]: I don't see the changes

File: src/App.tsx (hunk 5 above)
Comment (akash-dhande-ib) [reply src/App.tsx:36]: I don't see the changes

File: src/assets/scss/_common.scss (hunk 2 above)
Comment (amit-ib) [reply src/assets/scss/_common.scss:15]: I think code was not pushed properly kindly take a look now

**Pull request closed without merging**